pkg/operation/botanist/vpnseedserver.go

```diff
@@ -27,24 +27,37 @@ import (
 	"k8s.io/utils/pointer"
 )
 
-const (
-	// DefaultDiffieHellmanKey is the default diffie-hellmann key
-	DefaultDiffieHellmanKey = `-----BEGIN DH PARAMETERS-----
+var (
+	diffieHellmanKeyData = map[string][]byte{
+		"dh2048.pem": []byte(`-----BEGIN DH PARAMETERS-----
 MIIBCAKCAQEA7cBXxG9an6KRz/sB5uiSOTf7Eg+uWVkhXO4peKDTARzMYa8b7WR8
 B/Aw+AyUXtB3tXtrzeC5M3IHnuhFwMo3K4oSOkFJxatLlYKeY15r+Kt5vnOOT3BW
 eN5OnWlR5Wi7GZBWbaQgXVR79N4yst43sVhJus6By0lN6Olc9xD/ys9GH/ykJVIh
 Z/NLrxAC5lxjwCqJMd8hrryChuDlz597vg6gYFuRV60U/YU4DK71F4H7mI07aGJ9
 l+SK8TbkKWF5ITI7kYWbc4zmtfXSXaGjMhM9omQUaTH9csB96hzFJdeZ4XjxybRf
 Vc3t7XP5q7afeaKmM3FhSXdeHKCTqQzQuwIBAg==
 -----END DH PARAMETERS-----
-`
+`,
+		)}
+	diffieHellmanKeyChecksum string
 )
 
-var diffieHellmanKeyChecksum string
-
 // init calculates the checksum of the default diffie hellman key
 func init() {
-	diffieHellmanKeyChecksum = utils.ComputeChecksum(map[string][]byte{"dh2048.pem": []byte(DefaultDiffieHellmanKey)})
+	diffieHellmanKeyChecksum = utils.ComputeChecksum(diffieHellmanKeyData)
+}
+
+func (b *Botanist) getDiffieHellmanSecret() component.Secret {
+	data, checksum := diffieHellmanKeyData, diffieHellmanKeyChecksum
+	if secret := b.LoadSecret(v1beta1constants.GardenRoleOpenVPNDiffieHellman); secret != nil {
+		data, checksum = secret.Data, utils.ComputeSecretChecksum(secret.Data)
+	}
```

Comment:
Will this work? I don't see `StoreSecret` being called anywhere. If I am not missing something, similar `LoadSecret` call will always return `nil`.